classes/user/form/ContactForm.inc.php

```diff
@@ -35,7 +35,10 @@ public function __construct($user)
         // Validation checks for this form
         $this->addCheck(new \PKP\form\validation\FormValidatorEmail($this, 'email', 'required', 'user.profile.form.emailRequired'));
         $this->addCheck(new \PKP\form\validation\FormValidator($this, 'country', 'required', 'user.profile.form.countryRequired'));
-        $this->addCheck(new \PKP\form\validation\FormValidatorCustom($this, 'email', 'required', 'user.register.form.emailExists', [DAORegistry::getDAO('UserDAO'), 'userExistsByEmail'], [$user->getId(), true], true));
+        $this->addCheck(new \PKP\form\validation\FormValidatorCustom($this, 'email', 'required', 'user.register.form.emailExists', function ($email, $userId) {
+            $user = Repo::user()->dao->getByEmail($email);
+            return !$user || $user->getId() != $userId;
+        }, [$user->getId()], true));
```

Comment:
Although we don't yet have a API for adding/editing users, this is the sort of thing that should probably be captured in the Repository's `validate` method. Or, if that's not in use yet, we could remove the `validate` method from the Repository so that it's not presumed to be complete.

Reply:
(In general, the old forms don't use the validate methods -- they're only used in the REST API.)

Reply:
Whoops, I think I brought that `Repository::validate` in wholesale from the Announcement code and forgot to check through it. I've removed it for now.